src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -191,15 +182,6 @@ public static class Props
 
     private volatile long lastProcessedMessageAt = System.currentTimeMillis();
 
-    /**
-     * This property is initially set to {@code true} which means that we have no information about the other nodes.
-     * Once all nodes are on at least this node version, it becomes {@code false}, which means that we are not
-     * upgrading from the previous version (major, minor).
-     *
-     * This property and anything that checks it should be removed in 5.0
-     */
-    private volatile boolean upgradeInProgressPossible = true;
```

Comment:
FWIW, this has not been removed on 5.0 as far as I can see, and instead a new boolean was added, creating an obvious race condition, which is never mentioned so presumably unexpected :shrug:

Reply:
As you can see in the code, there were two mechanisms to report the min cluster version. I don't know the reason, perhaps an oversight. I think we should refactor that here and in OSS.